flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/functions/sql/ProctimeSqlFunction.java

```diff
@@ -37,16 +34,15 @@ public ProctimeSqlFunction() {
 		super(
 				"PROCTIME",
 				SqlKind.OTHER_FUNCTION,
-				ReturnTypes.explicit(new ProctimeRelProtoDataType()),
+				ReturnTypes.explicit(factory ->
```

Comment:
Why don't we keep align with legacy planner's ProctimeSqlFunction?

Reply:
Previously set to false causes sorts of inexplicable errors in `LookupJoinTest`.
The fundamental reason is that:
`TimeIndicatorRelDataType` assigns `isNullable` and overrides `equals`, but `BasicSqlType` do `computeDigest` at the constructor, so this leads to the inconsistency between `equals` and `Digest`, which leads to a variety of problems.
Fixed it in ff7c1e7
(Looks like `GenericRelDataType` has same problem.)

Reply:
Flink planner just use `SqlTypeName.TIMESTAMP` for return type

Reply:
The `ProctimeSqlFunction` of Flink planner is `ProctimeMaterializeSqlFunction` in blink.
`ProctimeSqlFunction` in blink is the "proctime()" method to access a proctime attribute.(blink introduce a new buildIn function named "proctime()")